django/db/models/window.py

```diff
@@ -0,0 +1,190 @@
+from django.db.models.expressions import (
+    BaseExpression, Expression, ExpressionList, OrderBy,
+)
+from django.utils.functional import cached_property
+
+__all__ = ['RowRange', 'ValueRange', 'Window']
+
+
+class Window(Expression):
+    template = '%(expression)s OVER (%(window)s)'
+    filterable = False
+
+    def __init__(self, expression, partition_by=None, order_by=None, frame=None, output_field=None):
+        self.partition_by = partition_by
+        self.order_by = order_by
+        self.frame = frame
+
+        if not getattr(expression, 'window_compatible', False):
+            raise ValueError('expression must be compatible in an over-clause')
+
+        if self.partition_by is not None:
+            if isinstance(self.partition_by, (tuple, list)):
+                self.partition_by = ExpressionList(*self.partition_by)
+            else:
+                self.partition_by = ExpressionList(self.partition_by)
+
+        if self.order_by is not None:
+            if not isinstance(self.order_by, (BaseExpression, OrderBy, list, tuple)):
+                raise ValueError('order_by must either be of type OrderBy or an Expression')
+            # assume OrderBy if not list or tuple
+            if isinstance(self.order_by, (list, tuple)):
+                self.order_by = ExpressionList(*self.order_by)
+
+        super().__init__(output_field=output_field)
+        self.source_expression = self._parse_expressions(expression)[0]
+
+    @cached_property
+    def contains_aggregate(self):
```

Comment:
I think `contains_aggregate` and `contains_over_clause` could be defined as class attributes for simplicity.